tools/alloc2cover/main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Fprintln(os.Stderr, "Usage: alloc2cover <input-file> <source-root> [module-path]")
+		fmt.Fprintln(os.Stderr, "  Reads TinyGo -print-allocs output and converts to cover.out format")
+		fmt.Fprintln(os.Stderr, "  source-root: path to the root of the source files")
+		fmt.Fprintln(os.Stderr, "  module-path: optional Go module path prefix (e.g., github.com/user/repo)")
+		fmt.Fprintln(os.Stderr, "  Heap allocations are marked as uncovered (red in coverage reports)")
```

Comment:
can we do this using Go's standard library flag package for automatic documentation generation and straightforward way of adding more options in the future?
```go
var (
			flagInputFile, flagSourceRoot, flagModulePath string
		)
		flag.StringVar(&flagInputFile, "i", "", "Input to alloc2cover is the output of print-allocs")
		flag.StringVar(&flagSourceRoot, "root", "", "path to the root of the source files")
		flag.StringVar(&flagModulePath, "mod", "", "Optional Go module path prefix (e.g., github.com/user/repo)")
```

Reply:
and checking inputs:
```go
if flagInputFile == "" || flagSourceRoot == "" {
			flag.PrintDefaults()
			log.Fatal("invalid flags")
		}